### PR TITLE
samples: nrf9160: lwm2m_client: Handle network failures

### PR DESCRIPTION
**Pull request by rlubos**
Handle network errors reported by the LwM2M engine. In case network
failure is reported, the application will stop the LwM2M client, and
try to reconnect to the network. The LwM2M operation will be resumed
when LTE reconnects.

### CODE REVIEW

**Comment by rlubos** [comment]
@martiles This implementation of fault handling is slightly different from what we've shared with the customer some time ago. I think though that the approach where we actually restart the LTE connection in case of issues will work out better than waiting for events with a timeout. At least on my side with current implementation, I could for instance remove the sim card for some time and the LwM2M connection would resume eventually. I don't have means to tinker with the antenna though.

**Review by lemrey — APPROVED** [review]
LGTM

File: samples/nrf9160/lwm2m_client/src/main.c [hunk 1]
```diff
@@ -356,11 +356,44 @@ static void rd_client_event(struct lwm2m_ctx *client,
...
+			lte_lc_offline();
+			k_sleep(K_SECONDS(60));
+		} else {
+			LOG_INF("Connected to LTE network");
+			ui_led_set_pattern(UI_LTE_CONNECTED);
+		}
+	} while (ret < 0);
+
+#if defined(CONFIG_LWM2M_QUEUE_MODE_ENABLED)
+	ret = lte_lc_psm_req(true);
```
Comment (jtguggedal) [inline samples/nrf9160/lwm2m_client/src/main.c:388]: I'd put the PSM request before do-while loop. It saves an extra TAU request after connection is established and in some networks that TAU request may be rejected for an extended period of time (several minutes with repeated TAU requests and rejections going back and forth).

Comment (rlubos) [reply samples/nrf9160/lwm2m_client/src/main.c]: Thanks for the tip, suggestion applied.

**Review by martiles — APPROVED** [review]
lgtm

**Pull request merged**